presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -0,0 +1,222 @@
+                initConfig -> setPrestoClusterManager(initConfig, DummyClusterManager.class.getName())));
+        RubixInitializer rubixInitializer = new RubixInitializer(
+                new CatalogName("catalog"),
+                rubixConfigInitializer,
+                configurationInitializer);
+        InternalNode coordinatorNode = new InternalNode(
+                "master",
+                URI.create("http://127.0.0.1:8080"),
+                UNKNOWN,
+                true);
+        TestingNodeManager nodeManager = new TestingNodeManager(
+                coordinatorNode,
+                ImmutableList.of());
+        rubixInitializer.initializeRubix(nodeManager);
+
+        // wait for rubix to start
+        Failsafe.with(
+                new RetryPolicy<>()
+                        .withDelay(Duration.ofSeconds(1))
+                        // unlimited attempts
+                        .withMaxAttempts(-1)
+                        .withMaxDuration(Duration.ofMinutes(1)))
+                .run(() -> checkState(rubixConfigInitializer.isCacheReady()));
+
+        // get rubix caching filesystem
+        configurationInitializer = new HdfsConfigurationInitializer(config, ImmutableSet.of(rubixConfigInitializer));
+        configuration = new HiveHdfsConfiguration(configurationInitializer, ImmutableSet.of(
+                (dynamicConfig, ignoredContext, ignoredUri) -> {
+                    // make sure a new FS is created with Rubix caching enabled
+                    setCacheKey(dynamicConfig, "rubix");
+                    dynamicConfig.set("fs.file.impl", CachingLocalFileSystem.class.getName());
+                }));
+        environment = new HdfsEnvironment(configuration, config, new NoHdfsAuthentication());
+
+        cachingFileSystem = environment.getFileSystem(context, path);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        try (Closer closer = Closer.create()) {
+            closer.register(() -> deleteRecursively(tempDirectory, ALLOW_INSECURE));
+            closer.register(() -> nonCachingFileSystem.close());
+            closer.register(() -> cachingFileSystem.close());
+            closer.register(LocalDataTransferServer::stopServer);
+        }
+    }
+
+    @Test
+    public void testCacheRead()
+            throws Exception
+    {
+        Path file = getPath("some_file");
+
+        writeFile(nonCachingFileSystem.create(file), "Hello world");
+
+        double beforeCachedReads = getStats().getCachedReads();
+
+        assertEquals(readFile(cachingFileSystem.open(file)), "Hello world");
+        assertEquals(readFile(cachingFileSystem.open(file)), "Hello world");
+
+        double afterCachedReads = getStats().getCachedReads();
+        assertGreaterThan(afterCachedReads, beforeCachedReads);
+    }
+
+    @Test
+    public void testFileSystems()
+    {
+        assertInstanceOf(nonCachingFileSystem, FilterFileSystem.class);
```

Comment:
feels like very much implementation detail.
Do we care what is he class name? I would drop it. You are testing if cache works in `testCacheRead` - and that is what we care about.

Reply:
removed